packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -0,0 +1,1584 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/scheduler.dart';
+import 'package:flutter/services.dart';
+
+import 'actions.dart';
+import 'basic.dart';
+import 'focus_manager.dart';
+import 'focus_scope.dart';
+import 'framework.dart';
+import 'gesture_detector.dart';
+import 'selection_container.dart';
+import 'text_editing_intents.dart';
+import 'text_selection.dart';
+
+const Set<PointerDeviceKind> _kLongPressSelectionDevices = <PointerDeviceKind>{
+  PointerDeviceKind.touch,
+  PointerDeviceKind.stylus,
+  PointerDeviceKind.invertedStylus,
+};
+
+
+/// A widget that introduces an area for user selections.
```

Comment:
Nit: Maybe say "selection" without the plural since there can only be one selection at a time?  Up to you though if you think plural is more clear.

Reply:
although one selection at a time, users can make multiple selections during the life time of this widget, so I think it makes more sense to keep it plural.

Reply:
True, I agree with that :+1: